meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClientCache.java

```diff
@@ -81,14 +112,100 @@ public int countDirectChildren(final String key) {
         throw new MetaClientException("Not implemented yet.");
     }
 
-    @Override
-    public List<T> get(List<String> keys) {
-        throw new MetaClientException("Not implemented yet.");
+    private void populateAllCache() {
+        // TODO: Concurrently populate children and data cache.
+        if (!_cacheClient.exists(_rootEntry)) {
+            LOG.warn("Root entry: {} does not exist.", _rootEntry);
+            // Let the other threads know that the cache is populated.
+            _initializedCache.countDown();
```

Comment:
Why count down here?

Reply:
otherwise populateAllCache will return without counting down the latch so the cacheEventChange handler thread will wait indefinitely. We can have it twice or have it in the connect method (from previous impl)